pymagicc/io/__init__.py

```diff
@@ -0,0 +1,613 @@
+                        "\x00", ""
+                    )
+                elif isinstance(value, list):
+                    clean_list = [v.strip(" \t\n\r").replace("\x00", "") for v in value]
+                    single_cfg[namelist_to_read][key] = [v for v in clean_list if v]
+                else:
+                    if not isinstance(value, Number):
+                        raise AssertionError("value is not a number: {}".format(value))
+
+                    single_cfg[namelist_to_read][key] = value
+            except AttributeError:
+                if isinstance(value, list):
+                    if not all([isinstance(v, Number) for v in value]):
+                        raise AssertionError(
+                            "List where not all values are numbers? " "{}".format(value)
+                        )
+
+                    single_cfg[namelist_to_read][key] = value
+                else:
+                    raise AssertionError(
+                        "Unexpected cause in out parameters conversion"
+                    )
+
+    return single_cfg
+
+
+def pull_cfg_from_parameters_out_file(
+    parameters_out_file, namelist_to_read="nml_allcfgs"
+):
+    """
+    Pull out a single config set from a MAGICC ``PARAMETERS.OUT`` file.
+
+    This function reads in the ``PARAMETERS.OUT`` file and returns a single file with
+    the config that needs to be passed to MAGICC in order to do the same run as is
+    represented by the values in ``PARAMETERS.OUT``.
+
+    Parameters
+    ----------
+    parameters_out_file : str
+        The ``PARAMETERS.OUT`` file to read
+
+    namelist_to_read : str
+        The namelist to read from the file.
+
+    Returns
+    -------
+    :obj:`f90nml.Namelist`
+        An f90nml object with the cleaned, read out config.
+
+    Examples
+    --------
+    >>> cfg = pull_cfg_from_parameters_out_file("PARAMETERS.OUT")
+    >>> cfg.write("/somewhere/else/ANOTHERNAME.cfg")
+    """
+    parameters_out = read_cfg_file(parameters_out_file)
+    return pull_cfg_from_parameters_out(
+        parameters_out, namelist_to_read=namelist_to_read
+    )
+
+
+def get_generic_rcp_name(inname):
+    """
+    Convert an RCP name into the generic Pymagicc RCP name
+
+    The conversion is case insensitive.
+
+    Parameters
+    ----------
+    inname : str
+        The name for which to get the generic Pymagicc RCP name
+
+    Returns
+    -------
+    str
+        The generic Pymagicc RCP name
+
+    Examples
+    --------
+    >>> get_generic_rcp_name("RCP3PD")
+    "rcp26"
+    """
+    # TODO: move into OpenSCM
+    mapping = {
+        "rcp26": "rcp26",
+        "rcp3pd": "rcp26",
+        "rcp45": "rcp45",
+        "rcp6": "rcp60",
+        "rcp60": "rcp60",
+        "rcp85": "rcp85",
+    }
+    try:
+        return mapping[inname.lower()]
+    except KeyError:
+        error_msg = "No generic name for input: {}".format(inname)
+        raise ValueError(error_msg)
+
+
+def to_int(x):
```

Comment:
is this used anywhere? (probably out of this PR's scope but maybe worth making an issue to either a) move it into utils or b) delete)

Reply:
It's used in MAGICCData. Will likely disappear when that is updated to use ScmRun